<a href="https://colab.research.google.com/github/LuccaMello7/ExploratoryAnalisis_Testing/blob/main/Analise_Exploratoria.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Exploratory Analysis of Aeronautical Occurrences in Civil Aviation

In [39]:
DATA_PATH = "https://raw.githubusercontent.com/carlosfab/dsnp2/master/datasets/ocorrencias_aviacao.csv"

#IMPORT RELEVANT PACKAGES
import pandas as pd
#IMPORT CSV FILE

df = pd.read_csv(DATA_PATH, engine = 'python')

#SEE THE FIRST INPUTS
df.head()

codigo_ocorrencia  ... ocorrencia_dia_extracao
0    201305055424986  ...              2018-08-28
1    201605160250139  ...              2018-08-28
2    201805021421302  ...              2018-08-28
3    201311259977425  ...              2018-08-28
4    201103187273112  ...              2018-08-28

[5 rows x 22 columns]

<h1>**DataFrame Summary**</h1>

Using the `info()` method in a variable you get the summary of all variables (columns) of your dataset.  Its types are a valueble information to address to your Variable Dictionary

In [40]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5208 entries, 0 to 5207
Data columns (total 22 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   codigo_ocorrencia               5208 non-null   int64  
 1   ocorrencia_classificacao        5208 non-null   object 
 2   ocorrencia_tipo                 5208 non-null   object 
 3   ocorrencia_tipo_categoria       5208 non-null   object 
 4   ocorrencia_tipo_icao            5208 non-null   object 
 5   ocorrencia_latitude             5208 non-null   float64
 6   ocorrencia_longitude            5208 non-null   float64
 7   ocorrencia_cidade               5208 non-null   object 
 8   ocorrencia_uf                   5208 non-null   object 
 9   ocorrencia_pais                 5208 non-null   object 
 10  ocorrencia_aerodromo            5208 non-null   object 
 11  ocorrencia_dia                  5208 non-null   object 
 12  ocorrencia_horario              52

In [41]:
df.ocorrencia_classificacao.unique()

array(['ACIDENTE', 'INCIDENTE GRAVE', 'INCIDENTE'], dtype=object)

In [42]:
df.ocorrencia_tipo.value_counts()

FALHA DO MOTOR EM VOO                                  682
ESTOURO DE PNEU                                        604
FALHA OU MAU FUNCIONAMENTO DE SISTEMA / COMPONENTE     574
PERDA DE CONTROLE NO SOLO                              492
OUTROS                                                 350
                                                      ... 
CAUSADO POR RICOCHETE                                    1
RELACIONADO COM SECURITY                                 1
MANOBRA ABRUPTA                                          1
ALARME FALSO DE FOGO OU DE SUPERAQUECIMENTO              1
CORTANTE DE VENTO / TEMPESTADE                           1
Name: ocorrencia_tipo, Length: 63, dtype: int64

The method above return the list of unique variables - sometimes we want to see the frequency variables are happening. For the `ocorrencia_classificacao` we can see each amount of occurrency, we can also use `value_counts()`.

To transform each cell above in percentage we need to divide the values for the amount of lines of our dataset. As we have seen this before, the amount of lines can be obtained by df.shape[0].

In [43]:
df.shape #size of dataset

(5208, 22)

In [44]:
df.shape[0] #number of inputs

5208

In [45]:
df.ocorrencia_tipo.value_counts() / df.shape[0] #percentage of relative values

FALHA DO MOTOR EM VOO                                  0.130952
ESTOURO DE PNEU                                        0.115975
FALHA OU MAU FUNCIONAMENTO DE SISTEMA / COMPONENTE     0.110215
PERDA DE CONTROLE NO SOLO                              0.094470
OUTROS                                                 0.067204
                                                         ...   
CAUSADO POR RICOCHETE                                  0.000192
RELACIONADO COM SECURITY                               0.000192
MANOBRA ABRUPTA                                        0.000192
ALARME FALSO DE FOGO OU DE SUPERAQUECIMENTO            0.000192
CORTANTE DE VENTO / TEMPESTADE                         0.000192
Name: ocorrencia_tipo, Length: 63, dtype: float64

In [24]:
df.index

RangeIndex(start=0, stop=5208, step=1)

<h1>**Filtering values**</h1>

There are a few main methods for retrieving and filtering values in a DataFrame, each with its own particular characteristics.

Instead of explaining them, something that seems confusing I will explain the following methods `loc` and `iloc` 

<h1>loc</h1>

loc`[<lines>, <columns>]`

This method can be used in these two situations:



1.   select **Lines** by **labels /index**
2.   select **Lines** based on a **boolean array**

suppose I need to extract the latitude and longitude of the place where I had the occurrence 201805021421302 or even "in-flight engine failure" (FALHA DO MOTOR EM VOO) event








In [54]:
df["ocorrencia_tipo"] == "FALHA DO MOTOR EM VOO"

0       True
1       True
2       True
3       True
4       True
        ... 
5203    True
5204    True
5205    True
5206    True
5207    True
Name: ocorrencia_tipo, Length: 5208, dtype: bool

In [57]:
# check only the 5 first inputs

df.loc[df["ocorrencia_tipo"] == "FALHA DO MOTOR EM VOO"].head()

codigo_ocorrencia  ... ocorrencia_dia_extracao
0    201305055424986  ...              2018-08-28
1    201605160250139  ...              2018-08-28
2    201805021421302  ...              2018-08-28
3    201311259977425  ...              2018-08-28
4    201103187273112  ...              2018-08-28

[5 rows x 22 columns]

<h1>iloc</h1>

The use of iloc is simpler, as one passes selects rows and columns using numbers, in the same order as they appear in the DataFrame:

loc`[<row_number>, <column_number>]`

I can select a range of interest. For example, to select the first 3 inputs and the first 4 columns, I would do it as follows:

In [59]:
df.iloc[0:3, 0:4]

codigo_ocorrencia  ...                          ocorrencia_tipo_categoria
0    201305055424986  ...                FALHA OU MAU FUNCIONAMENTO DO MOTOR
1    201605160250139  ...  COLISÃO COM OBSTÁCULO DURANTE A DECOLAGEM E POUSO
2    201805021421302  ...                                          AERÓDROMO

[3 rows x 4 columns]